src/Illuminate/Validation/Concerns/ValidatesAttributes.php

```diff
@@ -255,7 +255,7 @@ protected function compareDates($attribute, $value, $parameters, $operator)
         if (is_null($date = $this->getDateTimestamp($parameters[0]))) {
             $comparedValue = $this->getValue($parameters[0]);
 
-            if (! is_string($comparedValue) && ! is_numeric($comparedValue) && ! $comparedValue instanceof DateTimeInterface) {
+            if (! is_null($comparedValue) && ! is_string($comparedValue) && ! is_numeric($comparedValue) && ! $comparedValue instanceof DateTimeInterface) {
```

Comment:
This works, although the latter conditions are repeated above  as well, so I would probably extract a method:

```suggestion
            if (! is_null($comparedValue) && ! $this->hasDateTypes($comparedValue)) {
```